server.py

```diff
@@ -33,12 +33,24 @@ def index():
 
 @mqtt.on_connect()
 def handle_connect(client, userdata, flags, rc):
-    for c_id in CLIENT_IDS:
-        mqtt.subscribe('client/' + c_id)
-
+    print("connected")
+    mqtt.subscribe(constants.NEW_CLIENT_INITIALIZATION)
+    print("connected after sbuscribing")
 
 @mqtt.on_message()
 def handle_mqtt_message(client, userdata, message):
+    #Add a new client and subscribe to appropriate topic
+    if message.topic == constants.NEW_CLIENT_INITIALIZATION:
+        #client_id = message.payload.decode()
+        payload = json.loads(message.payload.decode())
+        client_id = payload.get("message", None)
+        CLIENT_IDS.add(client_id)
+        initialize_datablocks(client_id)
+        mqtt.subscribe('client/' + client_id)
+        #mqtt.publish('client/' + client_id, "server received from"+client_id)
+        return
```

Comment:
move this block before the `if client_name in CLIENT_IDS:` block,

payload already defined below.